govcd/nsxt_alb_clouds.go

```diff
@@ -0,0 +1,194 @@
+	if err != nil {
+		return nil, err
+	}
+
+	typeResponses := []*types.NsxtAlbCloud{{}}
+	err = client.OpenApiGetAllItems(apiVersion, urlRef, queryParameters, &typeResponses, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Wrap all typeResponses into NsxtAlbCloud types with client
+	wrappedResponses := make([]*NsxtAlbCloud, len(typeResponses))
+	for sliceIndex := range typeResponses {
+		wrappedResponses[sliceIndex] = &NsxtAlbCloud{
+			NsxtAlbCloud: typeResponses[sliceIndex],
+			vcdClient:    vcdClient,
+		}
+	}
+
+	return wrappedResponses, nil
+}
+
+// GetAlbCloudByName returns NSX-T ALB Cloud by name
+func (vcdClient *VCDClient) GetAlbCloudByName(name string) (*NsxtAlbCloud, error) {
+	queryParameters := copyOrNewUrlValues(nil)
+	queryParameters.Add("filter", "name=="+name)
+
+	albClouds, err := vcdClient.GetAllAlbClouds(queryParameters)
+	if err != nil {
+		return nil, fmt.Errorf("error reading NSX-T ALB Cloud with Name '%s': %s", name, err)
+	}
+
+	if len(albClouds) == 0 {
+		return nil, fmt.Errorf("%s could not find NSX-T ALB Cloud with Name '%s'", ErrorEntityNotFound, name)
+	}
+
+	if len(albClouds) > 1 {
+		return nil, fmt.Errorf("found more than 1 NSX-T ALB Cloud with Name '%s'", name)
+	}
+
+	return albClouds[0], nil
+}
+
+// GetAlbCloudById returns NSX-T ALB Cloud by ID
+//
+// Note. This function uses server side filtering instead of directly querying endpoint with specified ID because such
+// endpoint does not exist
+func (vcdClient *VCDClient) GetAlbCloudById(id string) (*NsxtAlbCloud, error) {
+
+	queryParameters := copyOrNewUrlValues(nil)
+	queryParameters.Add("filter", "id=="+id)
+
+	albCloud, err := vcdClient.GetAllAlbClouds(queryParameters)
+	if err != nil {
+		return nil, fmt.Errorf("error reading NSX-T ALB Cloud with ID '%s': %s", id, err)
+	}
+
+	if len(albCloud) == 0 {
+		return nil, fmt.Errorf("%s could not find NSX-T ALB Cloud by ID '%s'", ErrorEntityNotFound, id)
+	}
+
+	return albCloud[0], nil
+}
+
+// CreateAlbCloud creates NSX-T ALB Cloud
+func (vcdClient *VCDClient) CreateAlbCloud(albCloudConfig *types.NsxtAlbCloud) (*NsxtAlbCloud, error) {
+	client := vcdClient.Client
+	if !client.IsSysAdmin {
+		return nil, errors.New("handling NSX-T ALB Clouds require System user")
+	}
+
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointAlbCloud
+	minimumApiVersion, err := client.checkOpenApiEndpointCompatibility(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	urlRef, err := client.OpenApiBuildEndpoint(endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	returnObject := &NsxtAlbCloud{
+		NsxtAlbCloud: &types.NsxtAlbCloud{},
+		vcdClient:    vcdClient,
+	}
+
+	err = client.OpenApiPostItem(minimumApiVersion, urlRef, nil, albCloudConfig, returnObject.NsxtAlbCloud, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating NSX-T ALB Cloud: %s", err)
+	}
+
+	return returnObject, nil
+}
+
+// Update is not supported up to at least VCD 10.3 therefore this function remains commented
```

Comment:
This commented out block is strange, given that 10.3 is already out :)

Reply:
Ah, the meaning I had is 10.3 still does not support it. 

Changed to
> // Update is not supported in VCD 10.3 and older therefore this function remains commented